src/meta/src/controller/catalog.rs

```diff
@@ -672,7 +672,94 @@ impl CatalogController {
         Ok(tables)
     }
 
-    pub async fn list_object_dependencies(&self) -> MetaResult<Vec<PbObjectDependencies>> {
+    pub async fn list_all_object_dependencies(&self) -> MetaResult<Vec<PbObjectDependencies>> {
```

Comment:
I found it's a little tricky to parameterize it. Not sure if there's a better way to share common logic with `list_creating_object_dependencies`.

Reply:
Why? The only difference is the filter condition, right?

Doesn't this work?
```rust
let filter = 
  if include_creating {
     sink::Column::TargetTable.is_not_null()
  } else {
     streaming_job::Column::JobStatus
                    .eq(JobStatus::Created)
                    .and(sink::Column::TargetTable.is_not_null())
  }
```

Reply:
I wasn't sure if the types would align since `filter` just takes a `F: IntoCondition` parameter.
```
    fn filter<F>(mut self, filter: F) -> Self
    where
        F: IntoCondition,
    {
        self.query().cond_where(filter.into_condition());
        self
    }
```

Let me take a try.

Reply:
Looks like they all use `Expr` under the hood. Making the change.

Reply:
Fixed: https://github.com/risingwavelabs/risingwave/pull/19066/commits/aa7b8eb8eaa1e3dfed9edd2b15437c173f8cb60f